packages/wallet-service/src/api/wallet.ts

```diff
@@ -380,6 +381,65 @@ interface LoadResult {
   xpubkey: string;
 }
 
+/*
+ * This lambda will be started by a SNSMessage on the load failed SNS configured
+ * in serverless.yml. It will receive all wallet load failed events published on
+ * the loadWalletAsync DLQ SNS topic
+ */
+export const loadWalletFailed: Handler<SNSEvent> = async (event) => {
+  const logger = createDefaultLogger();
+  const records = event.Records;
+
+  try {
+    for (let i = 0; i < records.length; i++) {
+      const snsEvent = records[i].Sns;
+      const { RequestID, ErrorMessage } = snsEvent.MessageAttributes;
+
+      // Process each failed load wallet event
+      const loadEvent: LoadEvent = JSON.parse(snsEvent.Message) as unknown as LoadEvent;
+
+      if (!loadEvent.xpubkey) {
+        logger.error('Received wallet load fail message from SNS but no xpubkey received');
+        continue;
```

Comment:
[doubt] What are the possible causes for this situation?

Reply:
This possibility would be a problem on the lambda itself, right? Or an invoke on lambda without correct params, right?

Reply:
Asking because if it is a possible problem on the lambda itself, it would be nice to alert us to investigate, because it may impact other calls to the lambda.

Reply:
Or we should build awareness over this error logs, turn it more visible for us.

Reply:
I added this try/catch thinking on a potential database failure which would cause the wallet to never be set to `error`, requiring manual intervetion

But it might happen for other reasons, like a failed deploy with missing env variables

I updated the alert message, indicating that this is important. It is also a MAJOR alert, so the on-call should treat it with utmost importance

e9b1f6d03a56c145e170a74754afc7d9ab8f2b11


I will create an issue to write a runbook for this situation

Reply:
> I updated the alert message, indicating that this is important. It is also a MAJOR alert, so the on-call should treat it with utmost importance

You are not alerting anything here. You just continue to the next event record.

Reply:
@pedroferreira1 you're right, I added an alert in 78ba032aa169fa6018decded99e39e45651bb670